builders/txBuilder.go

```diff
@@ -131,26 +183,47 @@ func transactionToNodeTransaction(tx *data.Transaction) (*transaction.Transactio
 		return nil, ErrInvalidValue
 	}
 
+	var guardianAddrBytes, guardianSigBytes []byte
+	if len(tx.GuardianAddr) > 0 {
+		guardianAddrBytes, err = core.AddressPublicKeyConverter.Decode(tx.GuardianAddr)
+		if err != nil {
+			return nil, err
+		}
+
+		guardianSigBytes, err = hex.DecodeString(tx.GuardianSignature)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return &transaction.Transaction{
-		Nonce:     tx.Nonce,
-		Value:     valueBI,
-		RcvAddr:   receiverBytes,
-		SndAddr:   senderBytes,
-		GasPrice:  tx.GasPrice,
-		GasLimit:  tx.GasLimit,
-		Data:      tx.Data,
-		ChainID:   []byte(tx.ChainID),
-		Version:   tx.Version,
-		Signature: signaturesBytes,
-		Options:   tx.Options,
+		Nonce:             tx.Nonce,
+		Value:             valueBI,
+		RcvAddr:           receiverBytes,
+		SndAddr:           senderBytes,
+		GasPrice:          tx.GasPrice,
+		GasLimit:          tx.GasLimit,
+		Data:              tx.Data,
+		ChainID:           []byte(tx.ChainID),
+		Version:           tx.Version,
+		Signature:         signaturesBytes,
+		Options:           tx.Options,
+		GuardianAddr:      guardianAddrBytes,
+		GuardianSignature: guardianSigBytes,
 	}, nil
 }
 
-func (builder *txBuilder) createUnsignedMessage(arg data.ArgCreateTransaction) ([]byte, error) {
-	arg.Signature = ""
+func TransactionToUnsignedTx(tx *data.Transaction) *data.Transaction {
```

Comment:
this function can be private if we don`t intend to use it anywhere else
```suggestion
func transactionToUnsignedTx(tx *data.Transaction) *data.Transaction {
```

Reply:
done